cryptography/src/bls12381/benches/dkg_recovery.rs

```diff
@@ -9,58 +9,92 @@ use commonware_cryptography::{
 use commonware_utils::{quorum, set::Ordered};
 use criterion::{criterion_group, BatchSize, Criterion};
 use rand::{rngs::StdRng, SeedableRng};
-use std::{collections::BTreeMap, hint::black_box, iter};
+use std::{collections::BTreeMap, hint::black_box};
 
 // Configure contributors based on context
 #[cfg(not(full_bench))]
 const CONTRIBUTORS: &[u32] = &[5, 10, 20, 50];
 #[cfg(full_bench)]
 const CONTRIBUTORS: &[u32] = &[5, 10, 20, 50, 100, 250, 500];
+const CONCURRENCY: &[usize] = &[1, 4, 8];
 
 fn benchmark_dkg_recovery(c: &mut Criterion) {
     let mut rng = StdRng::seed_from_u64(0);
     for &n in CONTRIBUTORS {
         let t = quorum(n);
-        c.bench_function(&format!("{}/n={} t={}", module_path!(), n, t), |b| {
-            b.iter_batched(
-                || {
-                    let mut private_keys = (0..n)
-                        .map(|i| PrivateKey::from_seed(i as u64))
-                        .collect::<Vec<_>>();
-                    let me = private_keys.pop().unwrap();
-                    let me_pk = me.public_key();
-                    let dealers = private_keys
-                        .iter()
-                        .map(|sk| sk.public_key())
-                        .collect::<Ordered<_>>();
-                    let players = Ordered::from_iter(iter::once(me_pk.clone()));
-                    let round_info = Info::new(0, None, dealers, players).unwrap();
+        for &concurrency in CONCURRENCY {
+            c.bench_function(
+                &format!("{}/n={} t={} conc={}", module_path!(), n, t, concurrency),
```

Comment:
The rationale for only using `CONCURRENCY` on the `finalize()` loop is that we can amortize the setup time.